Task2/src/Task2.2/Polynomial.cs

```diff
@@ -0,0 +1,175 @@
+			for (int i = 0; i < firstPolynom.СoefPolynom.Length; ++i)
+			{
+				for (int j = 0; j < secondPolynom.СoefPolynom.Length; ++j)
+				{
+					сoefPolynomResult[i + j] += secondPolynom.СoefPolynom[i] * firstPolynom.СoefPolynom[j];
+				}
+			}
+			return PolynomResult;
+		}
+
+		/// <summary>
+		/// The method overrides the mathimatical "plus" operation for working with two polynomials
+		/// </summary>
+		/// <param name="firstPolynom"> the first polynomial</param>
+		/// <param name="secondPolynom">the second polynomial</param>
+		/// <returns>return polynomial's result</returns>
+		public static Polynomial operator +(Polynomial firstPolynom, Polynomial secondPolynom)
+		{
+			if (firstPolynom == null)
+			{
+				throw new NullReferenceException();
+			}
+			if (secondPolynom == null)
+			{
+				throw new NullReferenceException();
+			}
+
+			int degreePolynomResult = 0;
+			if (firstPolynom.СoefPolynom.Length > secondPolynom.СoefPolynom.Length)
+			{
+				degreePolynomResult = firstPolynom.СoefPolynom.Length;
+			}
+			else if (firstPolynom.СoefPolynom.Length <= secondPolynom.СoefPolynom.Length)
+			{
+				degreePolynomResult = firstPolynom.СoefPolynom.Length;
+			}
+			double[] koefPolynomResult = new double[degreePolynomResult];
+
+			Polynomial PolynomResult = new Polynomial(koefPolynomResult);
+
+			for (int i = 0; i < degreePolynomResult; i++)
+			{
+				PolynomResult.СoefPolynom[i] = firstPolynom.СoefPolynom[i] + secondPolynom.СoefPolynom[i];
+			}
+
+			return PolynomResult;
+		}
+
+		/// <summary>
+		/// The method overrides the mathimatical "minus" operation for working with two polynomials
+		/// </summary>
+		/// <param name="firstPolynom"> the first polynomial</param>
+		/// <param name="secondPolynom">the second polynomial</param>
+		/// <returns>return polynomial's result</returns>
+		public static Polynomial operator -(Polynomial firstPolynom, Polynomial secondPolynom)
+		{
+			if (firstPolynom == null)
+			{
+				throw new NullReferenceException();
+			}
+			if (secondPolynom == null)
+			{
+				throw new NullReferenceException();
+			}
+
+			int degreePolynomResult = 0;
+			if (firstPolynom.СoefPolynom.Length > secondPolynom.СoefPolynom.Length)
+			{
+				degreePolynomResult = firstPolynom.СoefPolynom.Length;
+			}
+			else if (firstPolynom.СoefPolynom.Length <= secondPolynom.СoefPolynom.Length)
+			{
+				degreePolynomResult = firstPolynom.СoefPolynom.Length;
+			}
+
+			double[] koefPolynomResult = new double[degreePolynomResult];
+
+			Polynomial PolynomResult = new Polynomial(koefPolynomResult);
+
+			for (int i = 0; i < degreePolynomResult; i++)
+			{
+				PolynomResult.СoefPolynom[i] = firstPolynom.СoefPolynom[i] - secondPolynom.СoefPolynom[i];
+			}
+
+			return PolynomResult;
+		}
+
+		public override bool Equals(object obj)
+		{
+			if (obj!=null)
+			{
+				if (this.GetType()==obj.GetType())
+				{
+					if (obj is Polynomial polynom)
+					{
+						return ((this.сoefPolynom == polynom.сoefPolynom));
```

Comment:
```suggestion
			return Enumerable.SequenceEqual(СoefPolynom,polinom.СoefPolynom);
```
Тут было неверно т.к у тебя массив, а массивы сравнивать нельзя так же как и не массивы. Нужно сравнивать каждый элемент с каждым типо 
Coef[0]==polimom.Coef[0];
Coef[1]==polimom.Coef[1];
Гуглится вот так https://stackoverflow.com/questions/4423318/how-to-compare-arrays-in-c
Я раньше этого не знал и делал костылем в тесткейсе